File: mvn-advent-of-code/src/test/java/advent/twenty_sixteen/Day09ExplosivesInCyberspaceTest.java
package advent.twenty_sixteen;

import advent.common.DailyProblem;
import org.junit.Test;

import static org.junit.Assert.*;

public class Day09ExplosivesInCyberspaceTest {
    @Test
    public void checkReal() {
        DailyProblem<Long, Long> dp = new Day09ExplosivesInCyberspace("/twenty_sixteen/Day09-ExplosivesInCyberspace-input.txt");

        assertEquals(70186, dp.getPart1Answer().longValue());
        assertEquals(10915059201, dp.getPart2Answer().longValue());
    }

    @Test
    public void checkExample1() {
        DailyProblem<Long, Long> dp = new Day09ExplosivesInCyberspace("/twenty_sixteen/Day09-ExplosivesInCyberspace-example#1.txt");

        assertEquals(57, dp.getPart1Answer().longValue());
        assertEquals(56, dp.getPart2Answer().longValue());
    }
}

File: scala-advent-of-code/src/test/scala/advent/twenty_sixteen/Day09ExplosivesInCyberspaceTest.scala
package advent.twenty_sixteen

import org.scalatest.{FlatSpec, Matchers}

class Day09ExplosivesInCyberspaceTest extends FlatSpec with Matchers {
  "2016 Day 9 - Input" should "calculate answers" in {
    val dp = new Day09ExplosivesInCyberspace("/twenty_sixteen/Day09-ExplosivesInCyberspace-input.txt")

    dp.part1Answer should be(70186L)
    dp.part2Answer should be(10915059201L)
  }

  "2016 Day 9 - Example #1" should "calculate answers" in {
    val dp = new Day09ExplosivesInCyberspace("/twenty_sixteen/Day09-ExplosivesInCyberspace-example#1.txt")

    dp.part1Answer should be(57L)
    dp.part2Answer should be(56L)
  }

}
